perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourceREMS.java

```diff
@@ -0,0 +1,157 @@
+	 * or by additionalues_1 in format: {extSourceName}|{extSourceClass}|{eppn}|0
+	 *
+	 * @param subjects subjects that will be filtered
+	 * @return List without filtered subjects
+	 * @throws InternalErrorException
+	 */
+	private List<Map<String, String>> filterNonExistingUsers(List<Map<String, String>> subjects) throws InternalErrorException {
+		List<Map<String, String>> existingSubjects = new ArrayList<>();
+
+		for (Map<String, String> subject : subjects) {
+			if(isExistingUser(subject)) {
+				existingSubjects.add(subject);
+			}
+		}
+
+		return existingSubjects;
+	}
+
+	/**
+	 * Checks if the given subject has a corresponding user in Perun by ues REMS
+	 * or by additionalues_1 in format: {extSourceName}|{extSourceClass}|{eppn}|0
+	 *
+	 * @param subject subject
+	 * @return true if the subject has, false otherwise
+	 * @throws InternalErrorException
+	 */
+	private boolean isExistingUser(Map<String, String> subject) throws InternalErrorException {
+		if (subject.isEmpty()) {
+			return false;
+		}
+		String login = subject.get("login");
+
+		PerunSession sess = getSession();
+
+		List<User> usersFromREMS = perunBl.getUsersManagerBl().getUsersByExtSourceTypeAndLogin(sess, ExtSourcesManager.EXTSOURCE_REMS, login);
+		if (usersFromREMS.size() > 0) {
+			return true;
+		}
+
+		String extSource = subject.get("additionalues_1");
+		if (extSource == null) {
+			log.error("Failed to get additionalues_1 from subject {}", subject);
+			throw new InternalErrorException("Failed to get additionalues_1 from subject: " + subject);
+		}
+
+		String[] extSourceSplit = subject.get("additionalues_1").split("\\|", 4);
+		if(extSourceSplit.length != 4) {
+			log.error("Ivalid format of additionalues_1. It should be '{extSourceName}|{extSourceClass}|{eppn}|0'. Actual: {}", extSource);
+			throw new InternalErrorException("Ivalid format of additionalues_1. It should be '{extSourceName}|{extSourceClass}|{eppn}|0', Actual: " + extSource);
+		}
+
+		String extSourceName = extSourceSplit[0];
+		String eppn = extSourceSplit[2];
+
+		try {
+			// try to find user by additionalues_1
+			perunBl.getUsersManagerBl().getUserByExtSourceNameAndExtLogin(sess, extSourceName, eppn);
+			return true;
+		} catch (ExtSourceNotExistsException | UserExtSourceNotExistsException e) {
+			log.error("Failed to get extSource with name '{}'", extSourceName);
+		} catch (UserNotExistsException e) {
+			return false;
+		}
+		return false;
+	}
+
+	private PerunSession getSession() throws InternalErrorException {
+		final PerunPrincipal pp = new PerunPrincipal("ExtSourceREMS", ExtSourcesManager.EXTSOURCE_NAME_INTERNAL, ExtSourcesManager.EXTSOURCE_INTERNAL);
+		try {
+			return perunBl.getPerunSession(pp, new PerunClient());
+		} catch (InternalErrorException e) {
+			log.error("Failed to get session for ExtSourceREMS.", e);
+			throw new InternalErrorException("Failed to get session for ExtSourceREMS.", e);
```

Comment:
Because thrown exception is also logged to the perun log file, you don't need to log it twice.